src/Slices/ResourceDetails/UI/Tabs/LogTab/View.spec.tsx

```diff
@@ -1,147 +1,144 @@
-import React, { act } from "react";
+import React from "react";
 import { MemoryRouter } from "react-router";
+import { QueryClientProvider } from "@tanstack/react-query";
 import { render, screen } from "@testing-library/react";
 import { userEvent } from "@testing-library/user-event";
 import { StoreProvider } from "easy-peasy";
-import { Either } from "@/Core";
-import { QueryResolverImpl, getStoreInstance } from "@/Data";
-import {
-  Resource,
-  DynamicQueryManagerResolverImpl,
-  StaticScheduler,
-  DeferredApiHelper,
-  dependencies,
-} from "@/Test";
+import { delay, HttpResponse } from "msw";
+import { http } from "msw";
+import { setupServer } from "msw/node";
+import { getStoreInstance } from "@/Data";
+import { dependencies } from "@/Test";
+import { testClient } from "@/Test/Utils/react-query-setup";
 import { DependencyProvider } from "@/UI/Dependency";
-import { ResourceLogsQueryManager, ResourceLogsStateHelper } from "@S/ResourceDetails/Data";
 import { ResourceLogs } from "@S/ResourceDetails/Data/Mock";
 import { View } from "./View";
 
 function setup() {
   const store = getStoreInstance();
-  const apiHelper = new DeferredApiHelper();
-  const resourceLogsStateHelper = ResourceLogsStateHelper(store);
-  const resourceLogsQueryManager = ResourceLogsQueryManager(
-    apiHelper,
-    resourceLogsStateHelper,
-    new StaticScheduler()
-  );
-
-  const queryResolver = new QueryResolverImpl(
-    new DynamicQueryManagerResolverImpl([resourceLogsQueryManager])
-  );
 
   const component = (
-    <MemoryRouter>
-      <DependencyProvider dependencies={{ ...dependencies, queryResolver }}>
-        <StoreProvider store={store}>
-          <View resourceId={Resource.id} />
-        </StoreProvider>
-      </DependencyProvider>
-    </MemoryRouter>
+    <QueryClientProvider client={testClient}>
+      <MemoryRouter>
+        <DependencyProvider dependencies={dependencies}>
+          <StoreProvider store={store}>
+            <View resourceId={"abc"} />
+          </StoreProvider>
+        </DependencyProvider>
+      </MemoryRouter>
+    </QueryClientProvider>
   );
 
   return {
     component,
-    apiHelper,
   };
 }
+describe("ResourceLogsView", () => {
+  const server = setupServer();
 
-test("GIVEN ResourceLogsView THEN shows resource logs", async () => {
-  const { component, apiHelper } = setup();
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
 
-  render(component);
+  test("GIVEN ResourceLogsView THEN shows resource logs", async () => {
+    server.use(
+      http.get("/api/v2/resource/abc/logs", () => {
+        delay(100);
```

Comment:
Based on what you mentionned in the meeting and struggling for the tests, 100mls really a fraction of a second, so might still result in a flake.